src/gamma_inc.jl

```diff
@@ -476,39 +467,30 @@ External links: [DLMF 8.11.2](https://dlmf.nist.gov/8.11.2)
 See also: [`gamma_inc(a,x,ind)`](@ref SpecialFunctions.gamma_inc)
 """
 function gamma_inc_asym(a::Float64, x::Float64, ind::Integer)
-    wk = zeros(30)
-    flag = false
     acc = acc0[ind + 1]
-    amn = a - 1.0
-    t = amn/x
-    wk[1] = t
-    loop = 2
-    for indx = 2:20
-       amn -= 1.0
-       t *= amn/x
-       if abs(t) <= 1.0e-3
-           loop = indx
-           flag = true
-           break
-       end
-       wk[indx] = t
-    end
-    if !flag
-        loop = 20
-    end
-    sm = t
-    while true
-       if abs(t) < acc
-           break
-       end
-       amn -= 1.0
-       t *= amn/x
-       sm += t
+
+    # compute first 21 terms
+    ts = cumprod(ntuple(i -> (a - i) / x, Val(21)))
+    last_t = findfirst(x -> abs(x) < 1.0e-3, ts)
+    last_t = last_t === nothing ? 20 : last_t - 1
+    next_t = last_t + 1
+
+    # sum the smaller terms directly
+    sm = t = ts[next_t]
+    amn = a - next_t
+    while abs(t) > acc
```

Comment:
```suggestion
    while abs(t) >= acc
```

Since: `!(abs(t) < acc)` => `abs(t) >= acc`